src/nrncvode/sptree.hpp

```diff
@@ -412,243 +375,90 @@ void splay(SPBLK* n, SPTREE<SPBLK>* q) {
         left->uplink = n;
     if (right != nullptr)
         right->uplink = n;
-    q->root = n;
+    root = n;
     n->uplink = nullptr;
+}
 
-} /* splay */
-
-
-/* Original file: spaux.cpp
-
-  spaux.cpp:  This code implements the following operations on an event-set
-  or priority-queue implemented using splay trees:
-
-  n = sphead( q )		n is the head item in q (not removed).
-  spdelete( n, q )		n is removed from q.
-
-  In the above, n and np are pointers to single items (type
-  SPBLK *); q is an event-set (type SPTREE *),
-  The type definitions for these are taken
-  from file sptree.h.  All of these operations rest on basic
-  splay tree operations from file sptree.cpp.
-
-  The basic splay tree algorithms were originally presented in:
-
-  Self Adjusting Binary Trees,
-  by D. D. Sleator and R. E. Tarjan,
-  Proc. ACM SIGACT Symposium on Theory
-  of Computing (Boston, Apr 1983) 235-245.
-
-  The operations in this package supplement the operations from
-  file splay.h to provide support for operations typically needed
-  on the pending event set in discrete event simulation.  See, for
-  example,
-
-  Introduction to Simula 67,
-  by Gunther Lamprecht, Vieweg & Sohn, Braucschweig, Wiesbaden, 1981.
-  (Chapter 14 contains the relevant discussion.)
-
-  Simula Begin,
-  by Graham M. Birtwistle, et al, Studentlitteratur, Lund, 1979.
-  (Chapter 9 contains the relevant discussion.)
-
-  Many of the routines in this package use the splay procedure,
-  for bottom-up splaying of the queue.  Consequently, item n in
-  delete and item np in all operations listed above must be in the
-  event-set prior to the call or the results will be
-  unpredictable (eg:  chaos will ensue).
-
-  Note that, in all cases, these operations can be replaced with
-  the corresponding operations formulated for a conventional
-  lexicographically ordered tree.  The versions here all use the
-  splay operation to ensure the amortized bounds; this usually
-  leads to a very compact formulation of the operations
-  themselves, but it may slow the average performance.
-
-  Alternative versions based on simple binary tree operations are
-  provided (commented out) for head, next, and prev, since these
-  are frequently used to traverse the entire data structure, and
-  the cost of traversal is independent of the shape of the
-  structure, so the extra time taken by splay in this context is
-  wasted.
-
-  This code was written by:
-  Douglas W. Jones with assistance from Srinivas R. Sataluri
-
-  Translated to C by David Brower, daveb@rtech.uucp
```

Comment:
We might need these comments for licensing. I also quite like that it provides a link to the literature.